x/auth/tx/builder.go

```diff
@@ -38,10 +39,19 @@ type builder struct {
 }
 
 var (
-	_ authsigning.Tx   = &builder{}
-	_ client.TxBuilder = &builder{}
+	_ authsigning.Tx             = &builder{}
+	_ client.TxBuilder           = &builder{}
+	_ ante.HasExtensionOptionsTx = &builder{}
+	_ ExtensionOptionsTxBuilder  = &builder{}
 )
 
+type ExtensionOptionsTxBuilder interface {
+	client.TxBuilder
+
+	SetExtensionOptions(...*codectypes.Any)
+	SetNonCriticalExtensionOptions(...*codectypes.Any)
```

Comment:
might be related to bez's comment: shouldn't we put this in the TxBuilder interface? The Std builder can have no-ops for these.

Reply:
We can do that too... although it should actually return an error for StdTx. Otherwise it's unexpected behavior because the getters will always return nil